### ISSUE TITLE
Connection to firebird is not closed if there is an error in attach function

### ISSUE THREAD

**Issue by ferserc1** [hgourvest/node-firebird#299]
When a connection error occurs, e.g. because of access permissions, the socket is not closed properly and open connections are left on the server. This can be checked by querying the `MON$ATTACHMENTS` table:

```sql
SELECT MON$REMOTE_PROCESS
FROM MON$ATTACHMENTS
WHERE MON$REMOTE_PROCESS='/usr/local/bin/node'
```

I have modified the code of the `attach` function to have access from outside the driver to the connection, and I have managed to solve the problem by doing a `detach` to the `db` object, which I can access from the connection:

**node-firebird:**

```js
exports.attach = function(options, callback) {

    var host = options.host || DEFAULT_HOST;
    var port = options.port || DEFAULT_PORT;
    var manager = options.manager || false;
    var cnx = this.connection = new Connection(host, port, function(err) {
            ...
    }, options);

    return cnx;  // Fix: get access to connection object outside node-firebird
};
```

```js
const conn = Firebird.attach(this._config, (err, db) => {
    this._conn.push(conn);
    if (err) {
        // manually detach the connection in case of error
        conn.db.detach();
        ...
```

I understand that `db.detach()` should be done automatically in case of error.

**Comment by mariuz** [comment]
Verified fixed on the current driver: a failed attach (e.g. wrong credentials) closes its socket and leaves no residue — `MON$ATTACHMENTS` count is identical before and after a failed attach on 2.12.0. The attach/connect error paths were reworked in the 2.x line (proper socket teardown plus a connect timeout for pooled attaches). Thanks for the diagnosis and the patch you carried — closing; reopen if you can still count leftover attachments on 2.x.
